modules/stochastic_tools/src/state_sim/model/events/StateSimEvent.C

```diff
@@ -0,0 +1,87 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#include "StateSimEvent.h"
+#include <string>
+#include <typeinfo>
+#include <unordered_map>
+//#include "NextObjID.h"
+
+unsigned int StateSimEvent::_counter = 0;
+
+StateSimEvent::StateSimEvent(std::string name, EVENT_TYPE_ENUM ev_type)
+  : StateSimBase(name),
+    _ev_type(ev_type),
+    _related_items()
+{
+  _id = ++_counter;
+}
+
+StateSimEvent::~StateSimEvent(){} //still required for virtual destructor
+
+void
+StateSimEvent::addRelatedItem(StateSimBase & item)
+{
+
+  std::unordered_map<int, StateSimBase *>::const_iterator iterator = _related_items.find(item.id());
+
+  if (typeid(iterator->second) != typeid(item)) //see if the items are the same type if not then there is an error.  This should never happen error by user.
```

Comment:
Let's not use RTTI here. Can we use normal inheritance to determine what to do instead?